tests/test_apt-cacher-ng-formula.py

```diff
@@ -2,27 +2,90 @@
 # -*- coding: utf-8 -*-
 
 """
-test_apt-cacher-ng-formula
+test_apt-cacher-ng_formula
 ----------------------------------
 
-Tests for `apt-cacher-ng-formula` module.
+Tests for `apt_cacher_ng_formula` module.
 """
 
 import unittest
+import salt.utils.templates
+import salt.template
+import salt.loader
+import salt.renderers.jinja
+import pprint
+import logging
+import sys
+logging.basicConfig(
+    stream=sys.stderr,
+    level=logging.DEBUG
+)
 
-from apt-cacher-ng-formula import apt-cacher-ng-formula
+import salt.renderers.jinja
+import salt.transport
+import pprint
+from mock import MagicMock
+import mock
 
 
-class TestApt-cacher-ng-formula(unittest.TestCase):
+
+
+class TestAptCacherNgFormula(unittest.TestCase):
 
     def setUp(self):
-        pass
 
-    def test_something(self):
-        pass
+        salt.renderers.jinja.__salt__= MagicMock()
+        salt.renderers.jinja.__grains__= MagicMock()
+        salt.renderers.jinja.__opts__= {
+            'cachedir' : "/tmp/cachedir",
+            'jinja_trim_blocks' : False,
+            'jinja_lstrip_blocks' : False,
+            'allow_undefined' : False,
+            'file_client' : 'remote',
+            'transport' : 'local', #'raet', 'zeromq' 
+            'id' : 'somerole',
+            '__role' : 'master', # minion
+            'pillar' : {},
```

Comment:
Cool. This would be useful to have in the core salt testing code.
On Jan 25, 2015 4:37 AM, "James Michael DuPont" <notifications@github.com>
wrote:

> In tests/test_apt-cacher-ng-formula.py
> <https://github.com/westurner/apt-cacher-ng-formula/pull/2#discussion_r23503452>
> :
>
> >
> > -    def test_something(self):
> > -        pass
> > +        salt.renderers.jinja.__salt__= MagicMock()
> > +        salt.renderers.jinja.__grains__= MagicMock()
> > +        salt.renderers.jinja.__opts__= {
> > +            'cachedir' : "/tmp/cachedir",
> > +            'jinja_trim_blocks' : False,
> > +            'jinja_lstrip_blocks' : False,
> > +            'allow_undefined' : False,
> > +            'file_client' : 'remote',
> > +            'transport' : 'local', #'raet', 'zeromq'
> > +            'id' : 'somerole',
> > +            '__role' : 'master', # minion
> > +            'pillar' : {},
>
> I have a pillar parser now working, see my latest test case
> https://github.com/h4ck3rm1k3/tomcat-formula/blob/master/tests/pillartest.py
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/westurner/apt-cacher-ng-formula/pull/2/files#r23503452>
> .
>